### PR TITLE
feat: Mint NFT UI

### PR DESCRIPTION
**Pull request by arsood**
**What changed? Why?**

- Improved Mint NFT page
- Bring Mint NFT page up to Figma spec
- Add common components for NextImage and ProgressBar

**Screenshots:**

_Desktop_

Screenshot 2024-02-02 at 12 21 31 AM

_Tablet_

Screenshot 2024-02-02 at 12 22 39 AM

_Mobile_

Screenshot 2024-02-02 at 12 23 08 AM

**How has it been tested?**

- Locally

### CODE REVIEW

File: template/src/pageComponents/mint/ContractDemo.tsx [hunk 1]
```diff
@@ -47,17 +49,41 @@ export default function MintContractDemo() {
     return <span className="text-xl">loading...</span>;
   }
 
+  // TODO: Retrieve this dynamically
+  const ethAmount = 0.0001;
```
Comment (arsood) [inline template/src/pageComponents/mint/ContractDemo.tsx:53]: @Sneh1999 what is the best way to get this amount?

File: template/src/pageComponents/mint/ContractDemo.tsx [hunk 2]
```diff
@@ -47,17 +49,41 @@ export default function MintContractDemo() {
...
+        >
+          Mint
         </button>
+
+        <div className="items-center md:flex">
+          <div className="w-full flex-shrink-0 flex-grow md:max-w-[70%]">
+            <ProgressBar percent={45} />
+          </div>
+          <div className="mt-2 w-full flex-shrink flex-grow-0 text-boat-footer-light-gray md:mt-0 md:text-right">
+            94/200 Minted
```
Comment (arsood) [inline template/src/pageComponents/mint/ContractDemo.tsx:84]: @Sneh1999 what is the best way to get these data?

**Comment by Zizzamia** [comment]
It's 1am @arsood is missing

Gum-Gum...

https://www.youtube.com/watch?v=VskI3Il88PU

File: template/src/components/NextImage/NextImage.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import Image from 'next/image';
+
+type NextImageProps = {
+  src: string;
+  altText: string;
+  className?: string;
+};
+
+export default function NextImage({ src, altText, className }: NextImageProps) {
```
Comment (Zizzamia) [inline template/src/components/NextImage/NextImage.tsx:9]: Side note for next PR, we need to start move the NFT Image stuff inside the OnchainKit folder.
As ideally we need to start building a NFT Media component we can abstract into the OnchainKit.

Comment (arsood) [reply template/src/components/NextImage/NextImage.tsx:9]: Good point, that makes sense.

**Review by Zizzamia — APPROVED** [review]
Let's have a Toast!!!!

**Comment by Zizzamia** [comment]
Another intersting point is to start using The Graph for some part of the experience.

**Comment by Zizzamia** [comment]
Merged this for now, and than with @Sneh1999 can cover the follow up questions.

**Pull request merged**